### PR TITLE
Ecdsa recreate pair from secret

### PR DESCRIPTION
**Pull request by DieMyst**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -14,7 +14,7 @@ val scalaV = scalaVersion := "2.12.5"
 
 val commons = Seq(
   scalaV,
-  version                   := "0.0.1",
+  version                   := "0.0.2",
```
Comment (c-solo) [inline build.sbt:17]: Let's up Cats version to 1.2.0.

File: hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala [hunk 2]
```diff
@@ -72,6 +73,24 @@ class Ecdsa(curveType: String, scheme: String, hasher: Option[Crypto.Hasher[Arra
...
 
+  def pairFromPrivate[F[_]: Monad](sk: Secret): EitherT[F, CryptoError, KeyPair] = {
+    for {
+      ecSpec ← EitherT.fromOption(
+        Option(ECNamedCurveTable.getParameterSpec(curveType)),
+        CryptoError("Parameter spec for the curve is not available.")
+      )
+      keyPair ← nonFatalHandling {
+        val hex = sk.value.toHex
+        val d = new BigInteger(hex, HEXradix)
```
Comment (c-solo) [inline hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala:90]: What is 'd', 'g', 'q'? Let's write full names.

File: hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala [hunk 3]
```diff
@@ -72,6 +73,24 @@ class Ecdsa(curveType: String, scheme: String, hasher: Option[Crypto.Hasher[Arra
         } yield keyPair
     }
 
+  def pairFromPrivate[F[_]: Monad](sk: Secret): EitherT[F, CryptoError, KeyPair] = {
```
Comment (c-solo) [inline hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala]: Let's write doc.

**Review by c-solo — CHANGES_REQUESTED** [review]

File: hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala (hunk 2 above)
Comment (DieMyst) [reply hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala:90]: this is points on a curve, this are full names :) I can add the comment about this

File: build.sbt (hunk 1 above)
Comment (DieMyst) [reply build.sbt:17]: cats-effect is `1.0.0-RC3` now

File: hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala [hunk 4]
```diff
@@ -72,6 +73,31 @@ class Ecdsa(curveType: String, scheme: String, hasher: Option[Crypto.Hasher[Arra
         } yield keyPair
     }
 
+  /**
+    * Create pair of keys from known private key.
```
Comment (alari) [inline hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala]: What happens if we create a keypair, recover keypair from secret part of it, and then compare public keys? Do they match? If they do (or don't), it worth noting in the doc comment.

File: hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala [hunk 5]
```diff
@@ -72,6 +73,31 @@ class Ecdsa(curveType: String, scheme: String, hasher: Option[Crypto.Hasher[Arra
...
+    * @param sk secret key
+    * @return key pair
+    */
+  def pairFromPrivate[F[_]: Monad](sk: Secret): EitherT[F, CryptoError, KeyPair] = {
```
Comment (alari) [inline hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala]: - It's not from private, it's from secret
- A method should be named with a verb
- Let's avoid using unnecessary curly braces

File: hashsign/jvm/src/test/scala/fluence/crypto/SignatureSpec.scala [hunk 6]
```diff
@@ -125,5 +126,16 @@ class SignatureSpec extends WordSpec with Matchers {
       //try to store key into previously created file
       storage.storeKeyPair(keys).attempt.unsafeRunSync().isLeft shouldBe true
     }
+
+    "restore key pair from private key" in {
```
Comment (alari) [inline hashsign/jvm/src/test/scala/fluence/crypto/SignatureSpec.scala]: You use the word "restore" here, let's use it in method's name as well?

**Review by alari — CHANGES_REQUESTED** [review]

File: hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala (hunk 2 above)
Comment (c-solo) [reply hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala:90]: That looks good, thank!

File: hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala (hunk 4 above)
Comment (c-solo) [reply hashsign/jvm/src/main/scala/fluence/crypto/ecdsa/Ecdsa.scala]: Create -> Creates

**Review by c-solo — APPROVED** [review]
Looks good for me, please fix @alari  remarks before the merge.

**Review by alari — APPROVED** [review]

**Pull request merged**